### ISSUE TITLE
NavigationItem.Link.ContentFileName is returning incorrectly

### ISSUE THREAD

**Issue by majora2007** [vers-one/EpubReader#45]
## Description
The library is returning the wrong text for ContentFileName. In the case for this epub, it should return "Text/chapter01.xhtml" while it is returning "Text/../Text/chapter01.xhml". I'm not sure where the extra relative path is coming from, given it's not in the XML. 

Epub structure:
![image](https://user-images.githubusercontent.com/735851/170345067-eadb3e44-a88f-4b58-a337-4d59734edc99.png)

Code:
```
var navItems = await book.GetNavigationAsync();
foreach (var navigationItem in navItems)
            {
                if (navigationItem.NestedItems.Count > 0)
                {
                   var nestedChapters = new List<BookChapterItem>();

                    foreach (var nestedChapter in navigationItem.NestedItems)
                    {
                        if (nestedChapter.Link == null) continue;

                        // BUG: nestedChapter.Link.ContentFileName -> Is returning "/Text/../Text/chapter01.xhtml" when it should be "Text/chapter01.xhtml"
                        var key = BookService.CleanContentKeys(nestedChapter.Link.ContentFileName);
                        if (mappings.ContainsKey(key))
                        {
                            nestedChapters.Add(new BookChapterItem()
                            {
                                Title = nestedChapter.Title,
                                Page = mappings[key],
                                Part = nestedChapter.Link.Anchor ?? string.Empty,
                                Children = new List<BookChapterItem>()
                            });
                        }
                    }

                    CreateToCChapter(navigationItem, nestedChapters, chaptersList, mappings);
                }
```

Toc.ncx:
```
<navPoint id="navPoint5">
      <navLabel>
        <text>Day 0: Backstory and the Bridal Wars</text>
      </navLabel>
      <content src="Text/chapter1.xhtml"/>
    </navPoint>
    <navPoint id="navPoint6">
      <navLabel>
        <text>Day 1, Morning: The Start of a Slow Life</text>
      </navLabel>
      <content src="Text/chapter2.xhtml"/>
    </navPoint>
```

Manifest:
```
<manifest>
    <item id="cover" href="Text/cover.xhtml" media-type="application/xhtml+xml"/>
    <item id="frontmatter1.xhtml" href="Text/frontmatter1.xhtml" media-type="application/xhtml+xml"/>
    <item id="frontmatter2.xhtml" href="Text/frontmatter2.xhtml" media-type="application/xhtml+xml"/>
    <item id="toc.xhtml" href="Text/toc.xhtml" media-type="application/xhtml+xml" properties="nav"/>
    <item id="prologue.xhtml" href="Text/prologue.xhtml" media-type="application/xhtml+xml"/>
    <item id="prologue2.xhtml" href="Text/prologue2.xhtml" media-type="application/xhtml+xml"/>
    <item id="insert1.xhtml" href="Text/insert1.xhtml" media-type="application/xhtml+xml"/>
    <item id="prologue2_1.xhtml" href="Text/prologue2_1.xhtml" media-type="application/xhtml+xml"/>
    <item id="chapter1.xhtml" href="Text/chapter1.xhtml" media-type="application/xhtml+xml"/>
    ...
```

## Sample EPUB file
The file is under copyright

## Additional context
This is a EPUB 2 document and I have tested on v3.1.1, v3.1.0 and it is not working.

**Comment by majora2007** [comment]
@vers-one Any ideas? I can send you the file if you need it as well.

**Comment by vers-one** [comment]
@majora2007, thank you for reporting this. Could you check whether this is EPUB 2 or EPUB 3 file? You can see the EPUB version by looking at the `version` attribute of the `package` XML node in the OPF file. If it's EPUB 3.x, then the *Toc.ncx* file will be ignored (some books include it for compatibility with the old readers that don't support EPUB 3) and the EPUB 3 navigation file will be used instead. EPUB 3 navigation file is the first item in the manifest with the `properties="nav"` which is *toc.xhtml* in your case.

If your file is EPUB 3, could you post a couple of `<li><a href="...">` tags from that file?

**Comment by vers-one** [comment]
EPUB 3 specification [requires](https://www.w3.org/publishing/epub32/epub-packages.html#confreq-nav-a-href-default) that `href` attribute values in the navigation XHTML file must "resolve" to the content documents from the manifest. I guess it means that they don't have to be exactly equal to the `href` values of the manifest's `item` nodes. Instead they just have to point to the same file after relative to absolute path conversion.

I'm planning to do the following:
1. Revert [this change](https://github.com/vers-one/EpubReader/pull/29) (I didn't notice that it was changing values in the schema entities which should remain exactly as they are in the schema files.)
2. Add the `ContentFilePathInEpubArchive` property to the `EpubNavigationItemLink` class, similar to how it's done for `EpubContentFile`: https://github.com/vers-one/EpubReader/blob/009976c3fa05a7d24d50e65a33958034cd832172/Source/VersOne.Epub/Entities/EpubContentFile.cs#L6
3. Perform the relative to absolute path conversion (so something like `OEBPS/Text` + `../Text/chapter01.xhtml` will be turned into `OEBPS/Text/chapter01.xhtml`) and assign the result to the `ContentFilePathInEpubArchive` property.

Once that's done, you'll be able to compare `nestedChapter.Link.ContentFilePathInEpubArchive` to `contentFile.FilePathInEpubArchive` to find the matching chapter in the navigation tree.

**Comment by majora2007** [comment]
Ooo, that would be great and very simple for me to implement on my end.

**Comment by vers-one** [comment]
@majora2007, I've made the change. Could you try it out and see if it works for you before I publish it to Nuget?

**Comment by majora2007** [comment]
I can, but I honestly don't know how to. I tried to build your project then paste the dll into my bin/, but it doesn't seem to have taken.

**Comment by vers-one** [comment]
No worries, let me publish it to Nuget then.

**Comment by vers-one** [comment]
Published as v3.1.2.

**Comment by majora2007** [comment]
I have validated it is working. Thanks again.
